### PR TITLE
Lvc 317

### PR DESCRIPTION
**Pull request by ChyzhTolik**
Added the ability to stop server

### CODE REVIEW

File: RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.cpp [hunk 1]
```diff
@@ -91,53 +91,6 @@ void SQLServer::InsertPhoto(const Photo& photo)
 	}
 }
 
-std::vector<SQLServer::PhotoType> SQLServer::GetAllPhotos()
```
Comment (ShostakOleh) [inline RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.cpp:94]: Why delete this function?

File: RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.cpp [hunk 2]
```diff
@@ -189,18 +142,6 @@ SQLServer::~SQLServer()
 	Disconnect();
 }
 
-void SQLServer::CreatePhotos(const std::vector<PhotoType>& photos)
```
Comment (ShostakOleh) [inline RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.cpp:192]: Why delete this function?

**Review by ShostakOleh — APPROVED** [review]

File: RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.cpp (hunk 2 above)
Comment (ChyzhTolik) [reply RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.cpp:192]: These functions were used for the previous version of the table, when photos were writen into the table as binaries

File: RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp [hunk 3]
```diff
@@ -58,37 +65,90 @@ bool SocketServer::BindListeningSocket()
...
+		{ 
+			SOCKET result;
+
+			result = accept(m_listen_socket, NULL, NULL);
+
+			ready = true;
+			return result;
+		});
+	
+	std::cout << "Waiting...     Press \'ESC\' to stop the Server\n" << std::flush;
```
Comment (marko-yaciv) [inline RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp:82]: If the server will run as a service, this output will not be written. Try to use Loger)

File: RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp [hunk 4]
```diff
@@ -58,37 +65,90 @@ bool SocketServer::BindListeningSocket()
...
+		{
+			server_is_up = false;
+
+			sockaddr_in clientService;
+			clientService.sin_family = AF_INET;
+			clientService.sin_addr.s_addr = inet_addr(m_ip.c_str());
+			clientService.sin_port = htons(std::stoi(m_port));
+			
+			mock_socket = socket(m_host_info->ai_family, m_host_info->ai_socktype, m_host_info->ai_protocol);
+			connect(mock_socket, (SOCKADDR*)&clientService, sizeof(clientService));
```
Comment (marko-yaciv) [inline RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp:100]: connect function is used for client sockets. Why have you used this function in the server part? Is this feature of "asio" sockets?

File: RealTimeFaceMaskDetector/TCPServer/WindowsService.cpp [hunk 5]
```diff
@@ -120,7 +120,15 @@ bool Service::Initialize(const unsigned short argc, const wchar_t* const argv[])
 
 void Service::StartServerWork(bool& is_listening_started)
 {
-	std::thread listen_multiple_clients([&]() {s_socket_server.StartListening(is_listening_started); });
+	std::thread listen_multiple_clients([&]() 
+		{
+			s_instance->s_instance->s_socket_server->StartListening(is_listening_started);
```
Comment (marko-yaciv) [inline RealTimeFaceMaskDetector/TCPServer/WindowsService.cpp]: Why do we need to get s_instance from s_instance?? How it should works?

File: RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp (hunk 3 above)
Comment (ChyzhTolik) [reply RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp:82]: most likely this feature will be deprecated

File: RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp (hunk 4 above)
Comment (ChyzhTolik) [reply RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp:100]: It was used to interrupt accept function

File: RealTimeFaceMaskDetector/TCPServer/WindowsService.cpp (hunk 5 above)
Comment (ChyzhTolik) [reply RealTimeFaceMaskDetector/TCPServer/WindowsService.cpp]: A typo. I corrected it

**Review by nullStandard — APPROVED** [review]

**Pull request merged**